src/main/java/apoc/schema/Schemas.java

```diff
@@ -117,6 +135,101 @@ private AssertSchemaResult createUniqueConstraint(Schema schema, String lbl, Str
         return new AssertSchemaResult(lbl, key).unique().created();
     }
 
+    public List<AssertSchemaResult> recreateIndexes(long allIndexesTimeout, long dropAllIndexesTimeout, long recreateAllIndexesTimeout) throws ExecutionException, InterruptedException, IllegalArgumentException, TimeoutException {
+        Schema schema = db.schema();
+        List<AssertSchemaResult> result = Collections.synchronizedList(new ArrayList<>());
```

Comment:
we can just use the results of the future of the 2nd call.
no need for synchronized collection to be passed in

Reply:
Is your suggestion that each call builds up a `List<AssertSchemaResult>` and then we combine them together afterwards?

Reply:
yes